### PR TITLE
modify chain2 script (add ivector, diagnosis log, compute_wer.sh), and add wer results for mini_librispeech and wsj

### PR DESCRIPTION
**Pull request by songmeixu**

### CODE REVIEW

**Comment by danpovey** [comment]
@songmeixu I just notice the option --groups-per-minibatch still exists.. it should be named --minibatch size.  You can do git grep for groups_per_minibatch and groups-per-minibatch to find where it is; can you please change that?
And thanks a lot for this!

**Comment by jtrmal** [comment]
two quick comments:
* not sure if you "re"introduce the cuda_cmd variable
* the error messages are not useful too much:
```
['steps/chain2/internal/get_train_schedule.py', '--frame-subsampling-factor=3', '--num-jobs-initial=2', '--num-jobs-final=5', '--num-epochs=4.0', '--dropout-schedule=', '--num-scp-files=2', '--frame-subsampling-factor=3', '--initial-effective-lrate=0.001
', '--final-effective-lrate=0.0001', '--schedule-out=exp/chaina/tdnn2c_sp/schedule.txt']
steps/chain2/train.sh: will train for 4.0 epochs = 6 iterations
steps/chain2/train.sh: Copying transition model
steps/chain2/train.sh: Num langs is 1
steps/chain2/train.sh: training, iteration 0 of 6, num-jobs is 2
steps/chain2/train.sh: error detected training on iteration 0
```
On mini-librispeech... So far I didn't find anything in logs

**Comment by jtrmal** [comment]
Also I feel like it would be nice, if the training script would print out at least some of the usual statistics the python script did:
```
steps/chain2/train.sh: will train for 4.0 epochs = 6 iterations
steps/chain2/train.sh: Copying transition model
steps/chain2/train.sh: Num langs is 1
steps/chain2/train.sh: training, iteration 0 of 6, num-jobs is 2
steps/chain2/train.sh: training, iteration 1 of 6, num-jobs is 3
steps/chain2/train.sh: training, iteration 2 of 6, num-jobs is 3
steps/chain2/train.sh: training, iteration 3 of 6, num-jobs is 4
steps/chain2/train.sh: training, iteration 4 of 6, num-jobs is 4
steps/chain2/train.sh: training, iteration 5 of 6, num-jobs is 5
steps/chain2/train.sh: doing model combination
```
instead of something like
```
2020-04-13 14:28:18,465 [steps/nnet3/chain/train.py:523 - train - INFO ] Iter: 10/33   Jobs: 3   Epoch: 4.00/20.0 (20.0% complete)   lr: 0.003786
2020-04-13 14:30:44,625 [steps/nnet3/chain/train.py:523 - train - INFO ] Iter: 11/33   Jobs: 3   Epoch: 4.50/20.0 (22.5% complete)   lr: 0.003574
2020-04-13 14:33:10,827 [steps/nnet3/chain/train.py:523 - train - INFO ] Iter: 12/33   Jobs: 3   Epoch: 5.00/20.0 (25.0% complete)   lr: 0.003374
2020-04-13 14:35:36,587 [steps/nnet3/chain/train.py:523 - train - INFO ] Iter: 13/33   Jobs: 3   Epoch: 5.50/20.0 (27.5% complete)   lr: 0.003185
```
I'm not saying it needs to copy the output format precisely, but it would be good if it can provide similar info.

File: egs/mini_librispeech/s5/local/chain2/tuning/run_tdnn_1a.sh [hunk 1]
```diff
@@ -45,6 +60,14 @@ egs_extra_right_context=5
 # to group multiple speaker together in some cases).
 chunks_per_group=4
```
Comment (danpovey) [inline egs/mini_librispeech/s5/local/chain2/tuning/run_tdnn_1a.sh:61]: I believe chunks_per_group is not used and should be removed.

File: egs/wsj/s5/local/chain2/tuning/run_tdnn_1i.sh [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+# easily be zero, since we're not using a recurrent topology, but we put in a
+# little extra context so that we have more room to play with the configuration
+# without re-dumping egs.
+egs_extra_left_context=5
+egs_extra_right_context=5
+
+# The number of chunks (of length: see $chunk_width above) that we group
+# together for each "speaker" (actually: pseudo-speaker, since we may have
+# to group multiple speaker together in some cases).
+chunks_per_group=4
```
Comment (danpovey) [inline egs/wsj/s5/local/chain2/tuning/run_tdnn_1i.sh:80]: same here, delete, plus comment

**Comment by danpovey** [comment]
I am merging this as I think it was basically tested.  Will try to get others to test it too.

**Comment by danpovey** [comment]
but @songmeixu there were a couple of comments that you could resolve via another PR.

**Pull request merged**